components/automate-cli/pkg/diagnostics/integration/iam_v2.go

```diff
@@ -106,6 +112,26 @@ func CreateIAMV2Diagnostic() diagnostics.Diagnostic {
 	}
 }
 
+// borrowed heavily from components/automate-cli/cmd/chef-automate/iam.go::runIAMVersionCmd
+func getIamMajorVersion() (policies_common.Version_VersionNumber, error) {
+	ctx := context.Background()
+	apiClient, err := apiclient.OpenConnection(ctx)
+	if err != nil {
+		return 0, status.Wrap(err, status.APIUnreachableError,
+			"Failed to create a connection to the API")
+	}
+
+	resp, err := apiClient.PoliciesClient().GetPolicyVersion(ctx, &policies_req.GetPolicyVersionReq{})
+	if err != nil {
+		return 0, status.Wrap(err, status.APIError, "Failed to retrieve IAM version")
+	}
+	return resp.Version.Major, nil
+}
+
 func init() {
```

Comment:
I'm afraid `init()` is rarely the correct move 😉 
I'd propse we do the following:
- add an `iam-v2` tag to diagnostic tests that require IAM v2 (here: https://github.com/chef/automate/pull/886/files#diff-78d181d2f500dd5af4a70ec53130c49cR52)
- use diagnostic filters for these, like https://github.com/chef/automate/blob/master/integration/tests/upgrade_current_master.sh#L4
- upgrade-to-v2 in the `integration/tests/*.sh` scripts that run the iam v2 diagnostic tests

Reply:
✔️ we've squashed that away now.